### PR TITLE
Big UI changes (full change list in description)

### PR DESCRIPTION
**Pull request by WSSDude420**
- Made Overlay key available only in VKBindings system to prevent duplication
- Moved Overlay key binding to Bindings widget
- First time launch now has modal popup where user needs to bind Overlay key
- Settings and Bindings widgets have now modal popups when there are unsaved changes
- Added option to disable removing of dead key bindings (good for debugging)
- Added toggle for ImGui Diagnostic Window widget
- Reworked Settings widget (has two tabs, one for patches and one for developer options)
- Reworked how widgets are enabled/disabled in Overlay
- Mods now do not draw on first launch until Overlay key is bound
- Created new helper widget for unapplied changes modal dialog
- Made BindWidget more general
- Modal dialogs will block LuaVM draw temporarily to prevent ImGui context breakage (beware that Update still runs! dont draw shit in Update please!)
- Added optional X offset argument to BindWidget
- Added option to enable ImGui assertions at will (default is on)
- Nicer formatting for headings of Bindings widget

Already merged from split PRs:
- Fixed issue with console history in Console widget (found by @Sombra0xCC )
- Fix single inputs in VKBindings
- Fix ImGui.TreePop unavailable in Lua
- Added optional X offset argument to BoolWidget
- Added DelayedSetTrapInputInImGui function to be used from within Overlay (needed because of design changes)
- More smaller internal things not listed here before but listed in respective PRs

### CODE REVIEW

**Comment by WSSDude420** [comment]
Splitting PR into smaller chunks ⌛

**Comment by WSSDude420** [comment]
OK, tested it, everything should work the same as before when this was gigantic x)

File: src/Options.cpp [hunk 1]
```diff
@@ -90,6 +83,10 @@ void Options::ResetToDefaults()
     PatchDisableVignette = false;
     PatchDisableBoundaryTeleport = false;
     PatchDisableWin7Vsync = false;
+
+    RemoveDeadBindings = true;
+    EnableImGuiAssertions = true;
```
Comment (yamashi) [inline src/Options.cpp:88]: Still not a fan of this being enabled by default, I don't see why users should receive assertion on things they can't repair, it's just going to generate massive support request for something we can't help with

File: src/VKBindings.h [hunk 2]
```diff
@@ -72,11 +72,12 @@ constexpr USHORT VKBC_MWHEELDOWN  { RI_MOUSE_WHEEL  | 0 };
 constexpr USHORT VKBC_MWHEELRIGHT { RI_MOUSE_HWHEEL | 1 };
 constexpr USHORT VKBC_MWHEELLEFT  { RI_MOUSE_HWHEEL | 0 };
 
+struct Options;
 struct Overlay;
 struct D3D12;
 struct VKBindings
 {
-    VKBindings(Paths& aPaths);
+    VKBindings(Paths& aPaths, const Options& aOptions);
```
Comment (yamashi) [inline src/VKBindings.h:80]: acOptions

File: src/VKBindings.h [hunk 3]
```diff
@@ -87,7 +88,7 @@ struct VKBindings
     static uint64_t EncodeVKCodeBind(VKCodeBindDecoded aVKCodeBindDecoded);
     static const char* GetSpecialKeyName(USHORT aVKCode);
     
-    void Load(const Overlay& aOverlay);
+    bool Load(const Overlay& aOverlay);
```
Comment (yamashi) [inline src/VKBindings.h:91]: acOverlay

File: src/VKBindings.h [hunk 4]
```diff
@@ -141,6 +142,7 @@ struct VKBindings
     bool m_initialized{ false };
 
     Paths& m_paths;
+    const Options& m_options;
```
Comment (yamashi) [inline src/VKBindings.h:145]: m_cOptions

**Review by yamashi — CHANGES_REQUESTED** [review]

**Pull request merged**